modules/gdscript/gdscript_compiler.cpp

```diff
@@ -402,7 +402,8 @@ GDScriptCodeGenerator::Address GDScriptCompiler::_parse_expression(CodeGen &code
 							String global_class_path = ScriptServer::get_global_class_path(identifier);
 							if (ResourceLoader::get_resource_type(global_class_path) == "GDScript") {
 								Error err = OK;
-								res = GDScriptCache::get_full_script(global_class_path, err);
+								// Should not need to pass p_owner since analyzer will already have done it.
+								res = GDScriptCache::get_shallow_script(global_class_path, err, parser->script_path);
```

Comment:
For those curious, this is the fix. The other stuff is just trying to ensure dependencies are properly added in the `GDScriptCache`.